cnd/src/swap_protocols/halight.rs

```diff
@@ -243,7 +263,25 @@ where
     })
 }
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Copy, Clone, Debug)]
 pub struct Params {
-    pub secret_hash: SecretHash,
+    pub ethereum_identity: EthereumIdentity,
+    pub ethereum_absolute_expiry: Timestamp,
+    // pub ethereum_amount: asset::Ether,
+    pub lightning_identity: identity::Lightning,
+    pub lightning_cltv_expiry: Timestamp,
+    pub lightning_amount: asset::Bitcoin,
+}
+
+impl Params {
```

Comment:
```suggestion
impl From<HanEtherereumHalightBitcoinCreateSwapParams> for Params {
```
Implement the `From` trait to have `Into` for free.